identity/src/keypair.rs

```diff
@@ -108,67 +108,115 @@ impl Keypair {
+    )]
     pub fn into_ed25519(self) -> Option<ed25519::Keypair> {
-        #[allow(deprecated)]
-        match self {
-            Keypair::Ed25519(inner) => Some(inner),
-            _ => None,
+        match self.try_into() {
+            Ok(k) => Some(k),
+            Err(_) => None,
         }
     }
 
+    #[cfg(feature = "ed25519")]
+    pub fn try_into_ed25519(self) -> Result<ed25519::Keypair, OtherVariantError> {
+        self.try_into()
+    }
+
     #[cfg(feature = "secp256k1")]
+    #[deprecated(
+        since = "0.2.0",
+        note = "This method name does not follow Rust naming conventions, use `Keypair::try_into_secp256k1` instead."
+    )]
     pub fn into_secp256k1(self) -> Option<secp256k1::Keypair> {
-        #[allow(deprecated)]
-        match self {
-            Keypair::Secp256k1(inner) => Some(inner),
-            _ => None,
+        match self.try_into() {
+            Ok(k) => Some(k),
+            Err(_) => None,
         }
     }
 
-    #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+    #[cfg(feature = "secp256k1")]
+    pub fn try_into_secp256k1(self) -> Result<secp256k1::Keypair, OtherVariantError> {
+        self.try_into()
+    }
+
+    #[cfg(feature = "rsa")]
+    #[deprecated(
+        since = "0.2.0",
+        note = "This method name does not follow Rust naming conventions, use `Keypair::try_into_rsa` instead."
+    )]
     pub fn into_rsa(self) -> Option<rsa::Keypair> {
-        #[allow(deprecated)]
-        match self {
-            Keypair::Rsa(inner) => Some(inner),
-            _ => None,
+        match self.try_into() {
+            Ok(k) => Some(k),
+            Err(_) => None,
         }
     }
 
+    #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+    pub fn try_into_rsa(self) -> Result<rsa::Keypair, OtherVariantError> {
+        self.try_into()
+    }
+
     #[cfg(feature = "ecdsa")]
+    #[deprecated(
+        since = "0.2.0",
+        note = "This method name does not follow Rust naming conventions, use `Keypair::try_into_ecdsa` instead."
+    )]
     pub fn into_ecdsa(self) -> Option<ecdsa::Keypair> {
-        #[allow(deprecated)]
-        match self {
-            Keypair::Ecdsa(inner) => Some(inner),
-            _ => None,
+        match self.try_into() {
+            Ok(k) => Some(k),
+            Err(_) => None,
         }
     }
 
+    #[cfg(feature = "ecdsa")]
+    pub fn try_into_ecdsa(self) -> Result<ecdsa::Keypair, OtherVariantError> {
+        self.try_into()
+    }
+
     /// Decode an keypair from a DER-encoded secret key in PKCS#8 PrivateKeyInfo
     /// format (i.e. unencrypted) as defined in [RFC5208].
     ///
     /// [RFC5208]: https://tools.ietf.org/html/rfc5208#section-5
     #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+    #[deprecated(
+        since = "0.2.0",
+        note = "Deprecated, use `rsa::Keypair::try_decode_pkcs8` or `rsa::Keypair::try_decode_der` and promote it into `Keypair` instead."
```

Comment:
We no longer have that one unless I am mistaken.

```suggestion
        note = "Deprecated, use `rsa::Keypair::try_decode_pkcs8` and promote it into `Keypair` instead."
```